### PR TITLE
Fix GH-2398

### PR DESCRIPTION
**Pull request by mysticmind**
Fix GH-2398

### CODE REVIEW

**Review by oskardudycz — COMMENTED** [review]
As we're near the next major release, then I think that it could be better to pass StoreOptions through `IMethodCallParser.Parse` method and `LinqParsing.BuildWhereFragment`. It should be able to get it in `WhereCallParser.VisitMethodCall` as it has StoreOptions. It's a bit more invasive but more explicit and logical than exposing it through Field Mapping. At least IMHO.

@jeremydmiller, thoughts?

**Comment by jeremydmiller** [comment]
I agree with Oskar about passing StoreOptions into IMethodParser in v6

**Comment by mysticmind** [comment]
okay, will make changes accordingly.

**Comment by mysticmind** [comment]
@jeremydmiller @oskardudycz I have made the changes as suggested, this PR is ready for review.

**Review by jeremydmiller — APPROVED** [review]
I think this is good as is, and it's the right call in the long run to pass in the read only view of the StoreOptions

File: src/Marten/Linq/Parsing/EnumerableContains.cs [hunk 1]
```diff
@@ -19,10 +19,10 @@ public bool Matches(MethodCallExpression expression)
                expression.Arguments.Single().IsValueExpression();
     }
 
-    public ISqlFragment Parse(IFieldMapping mapping, ISerializer serializer, MethodCallExpression expression)
+    public ISqlFragment Parse(IFieldMapping mapping, IReadOnlyStoreOptions options, MethodCallExpression expression)
```
Comment (oskardudycz) [inline src/Marten/Linq/Parsing/EnumerableContains.cs:22]: Nice, @mysticmind, good that you removed also the obsolete serialiser param :+1:

**Review by oskardudycz — APPROVED** [review]
Looks good to me :+1:

**Comment by jeremydmiller** [comment]
I think this is ready to go in.

**Pull request merged**